packages/atomic/src/components/commerce/atomic-commerce-recommendation-interface/atomic-commerce-recommendation-interface.tsx

```diff
@@ -0,0 +1,270 @@
+   *  - Refused to execute inline script because it violates the following Content Security Policy directive: [...]
+   *  - Refused to apply inline style because it violates the following Content Security Policy directive: [...].
+   * @example:
+   * ```html
+   * <script nonce="<YOUR_GENERATED_NONCE>">
+   *  import {setNonce} from '@coveo/atomic';
+   *  setNonce('<YOUR_GENERATED_NONCE>');
+   * </script>
+   * ```
+   */
+  @Prop({reflect: true}) public CspNonce?: string;
+
+  /**
+   * The commerce interface language.
+   */
+  @Prop({reflect: true}) public language = 'en';
+
+  /**
+   * Whether analytics should be enabled.
+   */
+  @Prop({reflect: true}) public analytics = true;
+
+  private i18nClone!: i18n;
+
+  private contextController!: Context;
+
+  public constructor() {
+    this.commonInterfaceHelper = new CommonAtomicInterfaceHelper(
+      this,
+      'CoveoAtomic'
+    );
+
+    if (!this.commonInterfaceHelper.engineIsCreated()) {
+      return;
+    }
+
+    this.contextController = buildContext(this.bindings.engine);
+  }
+
+  public connectedCallback() {
+    this.i18nClone = this.i18n.cloneInstance();
+    this.i18n.addResourceBundle = (
+      lng: string,
+      ns: string,
+      resources: object,
+      deep?: boolean,
+      overwrite?: boolean
+    ) => this.addResourceBundle(lng, ns, resources, deep, overwrite);
+  }
+
+  componentWillLoad() {
+    if (this.CspNonce) {
+      setNonce(this.CspNonce);
+    }
+    this.initAriaLive();
+  }
+
+  @Watch('analytics')
+  public toggleAnalytics() {
+    this.commonInterfaceHelper.onAnalyticsChange();
+  }
+
+  @Watch('language')
+  public updateLanguage() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+
+    this.contextController.setLanguage(this.language);
+
+    this.commonInterfaceHelper.onLanguageChange();
+  }
+
+  @Watch('iconAssetsPath')
+  public updateIconAssetsPath() {
+    this.store.set('iconAssetsPath', this.iconAssetsPath);
+  }
+
+  @Listen('atomic/initializeComponent')
+  public handleInitialization(event: InitializeEvent) {
+    this.commonInterfaceHelper.onComponentInitializing(event);
+  }
+
+  @Listen('atomic/scrollToTop')
+  public scrollToTop() {
+    const scrollContainerElement = document.querySelector(this.scrollContainer);
+    if (!scrollContainerElement) {
+      this.bindings.engine.logger.warn(
+        `Could not find the scroll container with the selector "${this.scrollContainer}". This will prevent UX interactions that require a scroll from working correctly. Please check the CSS selector in the scrollContainer option`
+      );
+      return;
+    }
+
+    scrollContainerElement.scrollIntoView({behavior: 'smooth'});
+  }
+
+  /**
+   * Initializes the connection with an already preconfigured [headless commerce engine](https://docs.coveo.com/en/headless/latest/reference/commerce/).
+   */
+  @Method() public initializeWithEngine(engine: CommerceEngine) {
```

Comment:
I only included the initializeWithEngine method on the atomic-commerce-recommendation-interface because otherwise we could end up with multiple engines / states, which I believe we should absolutely avoid.

It think it's OK to have the regular initialize method on the atomic-commerce-interface, though, because there should only be one PLP or search interface displayed at a time. However, there could be multiple recommendation interfaces.